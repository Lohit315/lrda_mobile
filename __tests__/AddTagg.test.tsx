import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import TagWindow from '../lib/components/tagging';
import moxios from 'moxios';


Enzyme.configure({ adapter: new Adapter() });


beforeAll(() => {
    jest.spyOn(console, 'log').mockImplementation(() => {});
    jest.spyOn(console, 'error').mockImplementation(() => {});
    moxios.install();
  });
  
  afterAll(() => {
    console.log.mockRestore();
    console.error.mockRestore();
    moxios.uninstall();
  });
  
  jest.mock('../lib/components/ThemeProvider', () => ({
    useTheme: () => ({
      theme: 'mockedTheme', 
    }),
  }));
  
describe('TagWindowTest1', () => {
  const mockTags = ['Tag1', 'Tag2', 'Tag3'];
  const mockSetTags = jest.fn();

  it('renders without crashing', () => {
    const wrapper = shallow(<TagWindow tags={mockTags} setTags={mockSetTags} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('displays input field and tags list', () => {
    const wrapper = shallow(<TagWindow tags={mockTags} setTags={mockSetTags} />);
    expect(wrapper.find('TextInput').exists()).toBe(true);
    expect(wrapper.find('SwipeListView').exists()).toBe(true);
  });
  
});

  
describe('TagWindowTest2', () => {
    const mockTags = ['1', '2','3'];
    const mockSetTags = jest.fn();
  
    it('renders without crashing', () => {
      const wrapper = shallow(<TagWindow tags={mockTags} setTags={mockSetTags} />);
      expect(wrapper).toMatchSnapshot();
    });

    it('handles tag deletion when swiping', () => {
        const wrapper = shallow(<TagWindow tags={mockTags} setTags={mockSetTags} />);
        const swipeListView = wrapper.find('SwipeListView');
        swipeListView.props().onRightAction('0', {});    
        expect(mockSetTags).toHaveBeenCalledWith(['2','3']);
      });
    });
  