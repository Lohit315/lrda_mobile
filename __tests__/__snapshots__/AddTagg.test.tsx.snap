// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagWindowTest1 renders without crashing 1`] = `ShallowWrapper {}`;
